packages/amplify-graphql-api-construct-tests/src/__tests__/restricted-field-auth/restricted-field-auth-gen2.test.ts

```diff
@@ -0,0 +1,440 @@
+import * as path from 'path';
```

Comment:
This is a rename & refactor of the "sql-only" test. Primary changes:
- Add new cases for heterogeneous data sources
- Add new case for homogeneous DDB relationships using Gen2 references style relationships
- Refactored test implementations to promote reuse.

Note that we can't use the implementations in the gen1 test, because they rely on the gen2 operations & API.ts.